src/ripple/rpc/impl/ServerHandlerImp.cpp

```diff
@@ -567,7 +567,7 @@ ServerHandlerImp::processRequest (Port const& port,
         if ((request.size () > RPC::Tuning::maxRequestSize) ||
             ! reader.parse (request, jsonOrig) ||
             ! jsonOrig ||
-            ! (jsonOrig.isObject () || jsonOrig.isArray()))
+            ! jsonOrig.isObject ())
```

Comment:
The `isArray` support was allowing malformed JSON through. Just removing it fixes the issue, but the changes below avoid relying on the default `null` entry behavior for JSON object instances.